packages/loader/container-loader/src/container.ts

```diff
@@ -816,26 +816,45 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             return;
         }
 
-        const previousContextState = await this.context.snapshotRuntimeState();
+        // By default, close the container and let the host reload.
+        // If hotSwapContext is true in the loader options, then try
+        // to reload the context without closing the container.
+        type ReloadState = { hotSwap: true; prevState: IRuntimeState } | { hotSwap: false };
+        let state: ReloadState = { hotSwap: false };
+        const loaderOptionsHotSwap = this.options.hotSwapContext as boolean ?? false;
```

Comment:
rather than hotSwap can we call the option somthing more descriptive like `reloadExisitingContextOnCodeProposal`

Reply:
also, let's add typing for loader level options, but keep an any indexer, as we pass it down to other layers

Reply:
Agree about the typings.. was sort of thinking about that too.

For the name, I can try to think of a better name than `hotSwapContext`, but `reloadExistingContextOnCodeProposal` is also a little confusing, since it doesn't reload the existing context, it disposes it and loads a new one without closing the container. I'll try to think of something better if "hot swap" is unclear.

Reply:
Added `ILoaderOptions` type for the 2nd comment.

For the name (1st comment), I'm having a hard time thinking of a better name here. I actually thinkg "hotSwapContext" is pretty accurate. "hot-swap" seems like the correct term, because it describes that the context should reload without closing the container. "context" seems correct as well, because it is aligned with the existing terminology of the events: "contextChanged" and "contextDisposed". So we are describing how to reload the context.

The only alternative I could think of would be "reloadContext" or "shouldReloadContext" or something along those lines, because the Container itself will not reload the context if hotSwap is false. If hotSwap is true, the container will reload the context, but if it is false, it will close itself. The only problem I have with this name is that there is the assumption that the host will reload the context in this case.

With the comment added to the type, it should also be more clear when reading the code.  Let me know if you think of a better name or have a good reason to avoid "hotSwapContext".

Reply:
i'm ok with this. comment makes it's clearer too.